src/arrow_parquet/uri_utils.rs

```diff
@@ -100,7 +111,43 @@ async fn get_s3_object_store(bucket_name: &str) -> AmazonS3 {
         aws_s3_builder = aws_s3_builder.with_allow_http(true);
     }
 
-    let aws_profile_name = std::env::var("AWS_PROFILE").unwrap_or("default".to_string());
+    // GUC overrides the environment variable for AWS_CONFIG_FILE.
+    // when the environment variable is not set, the default value
+    // (~/.aws/config) is used by object_store
+    if let Some(aws_config_file) = AWS_CONFIG_FILE.get() {
+        std::env::set_var(
+            "AWS_CONFIG_FILE",
+            aws_config_file
+                .to_str()
+                .expect("invalid GUC AWS_CONFIG_FILE CStr"),
+        );
+    }
+
+    // GUC overrides the environment variable for AWS_SHARED_CREDENTIALS_FILE.
+    // when the environment variable is not set, the default value
+    // (~/.aws/credentials) is used by object_store.
+    if let Some(aws_shared_credentials_file) = AWS_SHARED_CREDENTIALS_FILE.get() {
+        std::env::set_var(
```

Comment:
do not set environment variables as it could be unexpected for other extensions... there should be a way to configure object_store.